internal/app/orchestrator/orchestrator.go

```diff
@@ -90,16 +105,172 @@ func New(ctx context.Context, l log.Logger, mapper mapper.Mapper, upstreamClient
 //
 // Cancel is an optional function to release resources in the producer. If
 // provided, the consumer may call this function multiple times.
-func (c *orchestrator) CreateWatch(req gcp.Request) (chan gcp.Response, func()) {
-	// TODO implement.
-	return nil, nil
+func (o *orchestrator) CreateWatch(req gcp.Request) (chan gcp.Response, func()) {
+	ctx := context.Background()
+
+	o.downstreamResponseChannelsMu.Lock()
+	if o.downstreamResponseChannels[&req] == nil {
+		// If this is the first time we're seeing the request from the
+		// downstream client, initialize a channel to feed future responses.
+		o.downstreamResponseChannels[&req] = make(chan gcp.Response)
+	}
+	o.downstreamResponseChannelsMu.Unlock()
+
+	aggregatedKey, err := o.mapper.GetKey(req)
+	if err != nil {
+		// Can't map the request to an aggregated key. Log and continue to
+		// propagate the response upstream without aggregation.
+		o.logger.With("err", err).With("req", req).Error(ctx, "failed to get aggregated key")
+		// Mimic the aggregated key.
+		// TODO (insert issue) This key needs to be made more granular to
+		//      uniquely identify a request. User-specified defaults?
+		aggregatedKey = fmt.Sprintf("%s%s_%s", unaggregatedPrefix, req.GetNode().GetId(), req.GetTypeUrl())
```

Comment:
using `req.GetNode().GetId()` will cause many such keys. Thoughts?

Reply:
I'd like to open an issue and work on this unconventional path, "non-aggregated" path in a separate PR.  I think it needs some more discussion.

Reply:
Added https://github.com/envoyproxy/xds-relay/issues/56 and linked.